homeassistant/components/switcher_kis/services.yaml

```diff
@@ -0,0 +1,68 @@
+update_schedules:
+  description: 'Manually retrieve schedules data from the device.'
+
+turn_on_15_minutes:
```

Comment:
Why do we need 4 different services for this? Why not use a single service and pass in a time period?

Reply:
@rohankapoorcom 
There is actually a funny reason for it, Please take a look at the [device](https://switcher.co.il/%D7%9E%D7%95%D7%A6%D7%A8/%D7%A1%D7%95%D7%95%D7%99%D7%A6%D7%A8/#iLightbox[]/0).
As you can see it has like a small led circle in the middle indicating weather or not the device is on.
Around it has a bigger circle divided to 4 quarters indicating the timer state, 0-15 minutes, 16-30 minutes, 31-45 minutes and 46-60 minutes.

There can be 4 `turn_on` scenarios:
- The user calls the `turn_on` command: the middle led will get turned on and will turned off once the device reaches its time limit or the user manually called the `turn_off` command.

- The user calls the `turn_on` and sets a timer for more then 60 minutes (it can be 61 minutes for the example sake): unfortunately the result will be the same as the previous scenario with the exception of the device being turned off after 61 minutes or if the it reached its configured time limit or of course if the user manually turned it off. From the device's display perspective, it will make no use of the outer led circle because it can't display more then 60 minutes and it can only set the timer at the time of the request.

- The user calls the `turn_on` command with the timer of 15, 30, 45 or 60 minutes: again, the center led will get turned on and will get turned off when the device is turned off. The difference here will be that the outer circle will get lighted as well, and will decrease itself based on the timer state, for 46-60 minutes all 4 quarters will be on, for 31-45 only the first 3 quarters will be on and so on...

- The user calls the `turn_on` command and sets a timer to anything from 1 to 60 minutes and not coordinated with 15 minutes steps, for the example sake, let's say it was set to 17 minutes: since the 16-30 minutes qualify for two quarters to be turned on, that means that by display on the device, the user will see the device turned on for 30 minutes (for 2 minutes) and then it will be trimmed to one quarter which indicates 15 minutes.
Which is very annoying to my taste.

Eventually I figured I have to choose one of the following three approaches to handle this issue:
- Create 1 turn_on_with_timer service that accepts any timedelta as an argument.
And let the user figure out by himself the device's display conventions.

- Create 1 turn_on_with_timer that accepts only specific timedelta or integer values representing 15, 30, 45 or 60 minutes.
But I think this will be confusing and unintuitive for the common user.

- Create 4 parallel services, one for each 15 minutes step and have it take no argument.
Considering not a lot of users will ever turn on the device for more then 60 minutes, and the ones that will can easily accomplish that using the built-in schedules of HA's automation, eventually I've chose this approach.

It's not an issue for me my friend, all 4 services call the same method that just takes the timer from the service name.
Converting it to one service with a schema that accepts one argument for the timer, shouldn't take more then 15 minutes, half-hour tops.
I'll be happy to change it if you think it will be better for the common user.

I would be happy to hear you opinion on this issue.

Reply:
Removed, not relevant anymore.